packages/algorand-js/test/integration/deleteApp.ts

```diff
@@ -0,0 +1,54 @@
+import { SignType, SSCCallsParam, TransactionType } from "@algorand-builder/algob/build/types";
+import { assert } from "chai";
+
+import { ERRORS } from "../../src/errors/errors-list";
+import { Runtime, StoreAccountImpl } from "../../src/index";
+import { expectTealErrorAsync } from "../helpers/errors";
+import { getProgram } from "../helpers/files";
+import { useFixture } from "../helpers/integration";
+
+describe("Algorand Smart Contracts - Delete Application", function () {
+  useFixture("stateful");
+  const john = new StoreAccountImpl(1000);
+
+  let runtime: Runtime;
+  let program: string;
+  let deleteParams: SSCCallsParam;
+  const flags = {
+    sender: john.account,
+    globalBytes: 32,
+    globalInts: 32,
+    localBytes: 8,
+    localInts: 8
+  };
+  this.beforeAll(async function () {
+    runtime = new Runtime([john]); // setup test
+    program = getProgram('deleteApp.teal');
+
+    deleteParams = {
+      type: TransactionType.DeleteSSC,
+      sign: SignType.SecretKey,
+      fromAccount: john.account,
+      appId: 10,
+      payFlags: {},
+      appArgs: []
+    };
+  });
+
+  it("should fail during delete application if app id is not defined", async function () {
+    await expectTealErrorAsync(
+      async () => await runtime.executeTx(deleteParams, program, []),
+      ERRORS.TEAL.APP_NOT_FOUND
+    );
+  });
+
+  it("should delete application", async function () {
+    const appId = await runtime.addApp(flags, {}, program);
+    deleteParams.appId = appId;
+    await runtime.executeTx(deleteParams, program, []);
+
+    // verify app is deleted
+    const res = runtime.ctx.state.globalApps.has(appId);
+    assert.equal(res, false);
+  });
+});
```

Comment:
Please add one more test: 
+ create an app
+ execute transaction which should be rejected by logic (maybe we need to modify teal code)
+ assert that the app is not deleted as txn is rejected.

Reply:
thanks for suggestion